admin/src/containers/MainLayout/PanelContext.js

```diff
@@ -0,0 +1,53 @@
+import React, { Component, createContext } from 'react';
+import PropTypes from 'prop-types';
+
+const PanelContext = createContext({
+  component: null,
+  title: null,
+  props: {},
+  openPanel: () => {},
+  closePanel: () => {},
+  isPanelOpen: false
+});
+
+export class PanelProvider extends Component {
+  static propTypes = {
+    children: PropTypes.oneOfType([
+      PropTypes.element,
+      PropTypes.object,
+      PropTypes.arrayOf(PropTypes.element)
+    ])
+  };
+
+  openPanel = (component, title = null, props = {}) => {
+    this.setState({
+      component,
+      title,
+      isPanelOpen: true,
+      props
+    });
+  };
+
+  closePanel = () =>
+    this.setState({
+      component: null,
+      title: null,
+      isPanelOpen: false,
+      props: {}
+    });
+
+  state = {
```

Comment:
I usually put default state on the top of the component declaration

Reply:
I took this idea from an article:
https://medium.com/@BogdanSoare/how-to-use-reacts-new-context-api-to-easily-manage-modals-2ae45c7def81

![screen shot 2018-06-20 at 11 29 40](https://user-images.githubusercontent.com/1215868/41650021-551669a4-747d-11e8-8866-fb44a9717aa2.png)

I haven't tested it with state at the top, but it is likely to break.